types/wlr_screenshooter.c

```diff
@@ -0,0 +1,66 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <wayland-server.h>
+#include <wlr/types/wlr_screenshooter.h>
+#include <wlr/types/wlr_output.h>
+#include <wlr/util/log.h>
+#include "screenshooter-protocol.h"
+
+static void screenshooter_shoot(struct wl_client *client,
+		struct wl_resource *_screenshooter, uint32_t id,
+		struct wl_resource *_output, struct wl_resource *_buffer) {
+	struct wlr_screenshot *screenshot;
+	if (!(screenshot = calloc(1, sizeof(struct wlr_screenshot)))) {
+		return;
+	}
+	screenshot->output = _output;
+	screenshot->resource = wl_resource_create(client,
+		&orbital_screenshot_interface, wl_resource_get_version(_screenshooter), id);
+	wlr_log(L_DEBUG, "new screenshot %p (res %p)", screenshot, screenshot->resource);
+	wl_resource_set_implementation(screenshot->resource, NULL, screenshot, NULL);
+	// TODO: orbital_screenshot_send_done(screenshot->resource);
```

Comment:
When you do implement this, make sure you sync it up with the output's frame event.